source/agora/test/Base.d

```diff
@@ -907,6 +907,117 @@ public class TestAPIManager
             throw ex;
         }
     }
+
+    /***************************************************************************
+
+        Add blocks up to the provided height
+
+        This is a helper function to enable some common steps to getting blocks
+            externalized during the Network Unit tests
+        Params:
+            block_height = the desired block height
+            enrolled_height = the block height of the enrollments
+                for the participating validators
+            client_idxs = client indices for the participating validators
+
+    ***************************************************************************/
+
+    public void generateBlocks (Height block_height,
+        Height enrolled_height = Height(0),
+        string file = __FILE__, size_t line = __LINE__)
+    {
+        generateBlocks(iota(GenesisValidators), block_height,
+            enrolled_height, file, line);
+    }
+
+    /// Ditto
+    public void generateBlocks (Idxs)(Idxs client_idxs, Height block_height,
```

Comment:
Why don't you pass in `Clients` instead of `Idxs` as in other functions like `expectBlock`, and `wairForPreimages`. It would be good to make the related functions look consistent.

Reply:
Also indices